packages/editor/src/store/reducer.js

```diff
@@ -913,6 +913,38 @@ export function postLock( state = { isLocked: false }, action ) {
 	return state;
 }
 
+/**
+ * Post saving lock.
+ *
+ * When post saving is locked, the post cannot be published or updated.
+ *
+ * @param {PostSavingLockState} state  Current state.
+ * @param {Object}              action Dispatched action.
+ *
+ * @return {PostLockState} Updated state.
+ */
+export function postSavingLock( state = [], action ) {
+	const { lockName } = action;
+	switch ( action.type ) {
+		case 'LOCK_POST_SAVING':
```

Comment:
I don't have a strong preference, but tracking this as an object could simplify the reducer implementation a fair bit:

```js
case 'LOCK_POST_SAVING':
case 'UNLOCK_POST_SAVING':
	return {
		...state,
		[ lockName ]: action.type === 'LOCK_POST_SAVING'
	};
```

Reply:
ah, good suggestion, this would simplify the structure, i'll give that a try

Reply:
resolved in https://github.com/WordPress/gutenberg/pull/10649/commits/8fea9a0b1665ee73bec67bad457bfa5c4cc9f8d4